lib/postgrex/replication.ex

```diff
@@ -693,10 +717,11 @@ defmodule Postgrex.Replication do
   def handle_call({:start_replication, opts}, _from, %{repl_opts: nil, copy_opts: nil} = s) do
     statement = command(:start_replication, opts)
 
-    with {:ok, protocol} <- Protocol.handle_replication(statement, s.protocol),
+    with {:ok, slot, protocol} <- maybe_create_slot(opts, s.protocol),
+         {:ok, protocol} <- Protocol.handle_replication(statement, protocol),
          {:ok, protocol} <- Protocol.checkin(protocol) do
-      s = %{s | protocol: protocol, repl_opts: opts}
-      {:reply, :ok, s}
+      reply = (slot && {:ok, slot}) || :ok
```

Comment:
I wonder if we should do `{:ok, slot}` and then we return `{:ok, nil}` if there is no slot?

Reply:
I think it's a good option. At least it seems more consistent to me: always return `{:ok, created_slot}` and some case no slot is created/returned.